### PR TITLE
[Feature] support cubeMapping in metal

### PR DESCRIPTION
**Pull request by Mee-gu**
support cubeMapping in metal

### CODE REVIEW

File: cocos/renderer/backend/Texture.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ namespace
             case TextureFormat::MTL_ABGR4:
                 return byte(2);
             case TextureFormat::D24S8:
-                return byte(4);
+                return byte(8);
```
Comment (minggo) [inline cocos/renderer/backend/Texture.cpp:60]: why 8 bytes?

File: cocos/renderer/backend/metal/CommandBufferMTL.mm [hunk 2]
```diff
@@ -413,6 +413,9 @@ MTLCullMode toMTLCullMode(CullMode mode)
     for(const auto& iter : unifornInfo)
     {
         const auto& bindUniformInfo = iter.uniformInfo;
+        if(bindUniformInfo.location == -1)
```
Comment (minggo) [inline cocos/renderer/backend/metal/CommandBufferMTL.mm]: may be add `UniforInfo.isValid()` is better

File: cocos/renderer/backend/Texture.cpp (hunk 1 above)
Comment (Mee-gu) [reply cocos/renderer/backend/Texture.cpp:60]: currently use MTLPixelFormatDepth32Float_Stencil8 as DepthStencil combined format, its 64 bits

**Comment by Mee-gu** [comment]
update review

File: cocos/renderer/backend/Texture.cpp [hunk 3]
```diff
@@ -55,7 +55,12 @@ namespace
             case TextureFormat::MTL_ABGR4:
                 return byte(2);
             case TextureFormat::D24S8:
+#if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS)
```
Comment (minggo) [inline cocos/renderer/backend/Texture.cpp:58]: What about mac?

File: cocos/renderer/backend/metal/Utils.mm [hunk 4]
```diff
@@ -61,7 +66,7 @@
             return MTLPixelFormatA8Unorm;
             // Not all devices supports MTLPixelFormatDepth24Unorm_Stencil8, so change to MTLPixelFormatDepth32Float_Stencil8.
         case TextureFormat::D24S8:
-            return MTLPixelFormatDepth32Float_Stencil8;
+            return DEPTH_STENCIL_ATTACHMENT_PIXEL_FORMAT;
```
Comment (minggo) [inline cocos/renderer/backend/metal/Utils.mm:70]: Should modify the comments too.

File: cocos/renderer/backend/Texture.cpp (hunk 3 above)
Comment (Mee-gu) [reply cocos/renderer/backend/Texture.cpp:58]: mac use 32 bits combined format: MTLPixelFormatDepth24Unorm_Stencil8

**Comment by Mee-gu** [comment]
update reivew

**Pull request merged**